benchmarks/backend_request_func.py

```diff
@@ -260,7 +267,9 @@ async def async_request_openai_completions(
                             latency = time.perf_counter() - st
                         else:
                             body = json.loads(chunk)
-                            generated_text += body["choices"][0]["text"]
```

Comment:
This change will only work for the chat completion chunk object, but not the completion object. 

See reference: https://platform.openai.com/docs/api-reference/completions/object#completions/object-choices and https://platform.openai.com/docs/api-reference/chat/streaming

Reply:
Thanks for the pointers. Yes, I am trying to add another api

Reply:
I added an new backend parameter as `openai-chat` to work with endpoint `/v1/chat/completions`

```
python benchmark_serving.py --backend openai-chat --base-url https://api.openai.com --endpoint /v1/chat/completions --num-prompts 1 --model gpt-3.5-turbo --tokenizer openai-community/gpt2 --dataset ShareGPT_V3_unfiltered_cleaned_split.json --save-result
Namespace(backend='openai-chat', version='N/A', base_url='https://api.openai.com', host='localhost', port=8000, endpoint='/v1/chat/completions', dataset='ShareGPT_V3_unfiltered_cleaned_split.json', model='gpt-3.5-turbo', tokenizer='openai-community/gpt2', best_of=1, use_beam_search=False, num_prompts=1, request_rate=inf, seed=0, trust_remote_code=False, disable_tqdm=False, save_result=True)
  0%|                                                           | 0/1 [00:00<?, ?it/s]Traffic request rate: inf
100%|███████████████████████████████████████████████████| 1/1 [00:00<00:00,  1.30it/s]
Successful requests: 1
Benchmark duration: 0.769358 s
Total input tokens: 486
Total generated tokens: 25
Request throughput: 1.30 requests/s
Input token throughput: 631.70 tokens/s
Output token throughput: 32.49 tokens/s
Mean TTFT: 439.39 ms
Median TTFT: 439.39 ms
P99 TTFT: 439.39 ms
Mean TPOT: 30.72 ms
Median TPOT: 30.72 ms
P99 TPOT: 30.72 ms
```